monitoring/mongodb/dashboard.py

```diff
@@ -0,0 +1,1359 @@
+        core.StatValueMappingItem(text="STARTUP2", mapValue="5", color="", index=9),
+        core.StatValueMappingItem(text="UNKNOWN", mapValue="6", color="", index=8),
+        core.StatValueMappingItem(
+            text="ARBITER", mapValue="7", color="#8B8000", index=1
+        ),
+        core.StatValueMappingItem(text="DOWN", mapValue="8", color="", index=7),
+        core.StatValueMappingItem(text="ROLLBACK", mapValue="9", color="", index=6),
+        core.StatValueMappingItem(text="REMOVED", mapValue="10", color="", index=5),
+        core.StatValueMappingItem(
+            text="Exporter is not connected", mapValue="null", color="#FF7383", index=2
+        ),
+    )
+]
+
+
+class MongoDBMetrics:
+    """MongoDB metrics expressions"""
+
+    @staticmethod
+    def up(job_pattern):
+        return f'sum(mongodb_up{{job="{job_pattern}", namespace="{NAMESPACE}"}})'
+
+    @staticmethod
+    def version_info():
+        return f'count by (mongodb) (mongodb_version_info{{job=~"{Jobs.shards}|{Jobs.configsvr}|{Jobs.mongos}", namespace="{NAMESPACE}"}}) > 0'
+
+    @staticmethod
+    def db_stats(metric, filters=""):
+        base_filter = (
+            f'rs_nm=~"$rs_nm", rs_state="{RS_STATE_PRIMARY}", namespace="{NAMESPACE}"'
+        )
+        if filters:
+            base_filter += f", {filters}"
+        return f"max(mongodb_dbstats_{metric}{{{base_filter}}})"
+
+    @staticmethod
+    def replset_state(job_pattern):
+        return f'max by (pod) (mongodb_mongod_replset_my_state{{job=~"{job_pattern}", namespace="{NAMESPACE}"}})'
+
+    @staticmethod
+    def oplog_window():
+        return f"""min(
+  mongodb_mongod_replset_oplog_head_timestamp{{job=~"{Jobs.shard_pattern}", namespace="{NAMESPACE}"}}
+  -
+  mongodb_mongod_replset_oplog_tail_timestamp{{job=~"{Jobs.shard_pattern}", namespace="{NAMESPACE}"}}
+)"""
+
+    @staticmethod
+    def disk_usage():
+        return f'avg(mongodb_dbstats_fsUsedSize{{namespace="{NAMESPACE}"}}) / avg(mongodb_dbstats_fsTotalSize{{namespace="{NAMESPACE}"}})'
+
+    @staticmethod
+    def cursors_open():
+        return f'sum by (job) (mongodb_mongod_metrics_cursor_open{{rs_nm=~"$rs_nm", namespace="{NAMESPACE}"}})'
+
+    @staticmethod
+    def connections(conn_type, pod_pattern):
+        return f'sum by(pod) (mongodb_ss_connections{{namespace="{NAMESPACE}", conn_type="{conn_type}", pod=~"{pod_pattern}"}})'
+
+    @staticmethod
+    def opcounters(op_types, rate_interval="[$__rate_interval]"):
+        return f'sum by(job) (rate(mongodb_ss_opcounters{{legacy_op_type=~"{op_types}", rs_nm=~"$rs_nm", job=~"{Jobs.shard_pattern}", namespace="{NAMESPACE}"}}{rate_interval}))'
+
+    @staticmethod
+    def replication_lag():
+        return f'last_over_time(mongodb_mongod_replset_member_replication_lag{{job=~"{Jobs.shard_pattern}", namespace="{NAMESPACE}"}}[$__rate_interval]) > 0'
+
+
+def mongodb_stat(title, expr="", unit="none", thresholds=None, **kwargs):
```

Comment:
best to keep the underlying field name, to avoid confusion?
```suggestion
def mongodb_stat(title, expr="", format="none", thresholds=None, **kwargs):
```

....and then it may be removed from the prototype, and just rely on `kwargs` (and possibly a `defaults` value)

Reply:
same for thresholds